wrappers/react/src/hotTableClass.tsx

```diff
@@ -220,39 +237,54 @@ class HotTableClass extends React.Component<HotTableProps, {}> {
    * @param {React.ReactElement} rendererElement React renderer component.
    * @returns {Handsontable.renderers.Base} The Handsontable rendering function.
    */
-  getRendererWrapper(rendererElement: React.ReactElement): typeof Handsontable.renderers.BaseRenderer | any {
+  getRendererWrapper(rendererElement: React.ReactElement): typeof Handsontable.renderers.BaseRenderer {
     const hotTableComponent = this;
 
     return function (instance, TD, row, col, prop, value, cellProperties) {
       const renderedCellCache = hotTableComponent.getRenderedCellCache();
+      const portalContainerCache = hotTableComponent.getPortalContainerCache()
+      const key = `${row}-${col}`;
+
+       // Handsontable.Core type is mising guid
+      const instanceGuid = (instance as unknown as { guid: string }).guid;
+
+      const portalContainerKey = `${instanceGuid}${key}`
+      const portalKey = `${key}-${instanceGuid}`
 
-      if (renderedCellCache.has(`${row}-${col}`)) {
-        TD.innerHTML = renderedCellCache.get(`${row}-${col}`).innerHTML;
+      if (renderedCellCache.has(key)) {
+        TD.innerHTML = renderedCellCache.get(key).innerHTML;
       }
 
       if (TD && !TD.getAttribute('ghost-table')) {
-
-        const {portal, portalContainer} = createPortal(rendererElement, {
-          TD,
-          row,
-          col,
-          prop,
-          value,
-          cellProperties,
-          isRenderer: true
-        }, TD.ownerDocument);
+        const cachedPortal = hotTableComponent.portalCacheArray.find(portal => portal.key === portalKey);
```

Comment:
I agree with your conclusions, the `find` performance impact is not a concern on a tiny virtual set of data but for `renderAll` it would become a problem.

will make the change.